### PR TITLE
feat: add development server 

### PR DESCRIPTION
**Pull request by sumwatshade**
This will allow users to run a development server of this module federation use case via

```
yarn ibazel run //:dev

### CODE REVIEW

File: src/client/BUILD.bazel [hunk 1]
```diff
@@ -10,4 +10,5 @@ pkg_web(
         "%s/routes/routes/" % package_name(),
         "%s/host/host" % package_name(),
     ],
+    visibility = ["//visibility:public"],
```
Comment (Aghassi) [inline src/client/BUILD.bazel]: why does this need to be public?

File: src/client/routes/BUILD.bazel [hunk 2]
```diff
@@ -20,3 +20,13 @@ generate_route_manifest(
...
 )
+
+js_library(
+    name = "__pkg__",
+    package_name = "@carto/routes",
+    srcs = [
+        "index.js",
+        ":route_manifest",
+    ],
+    visibility = ["//visibility:public"],
```
Comment (Aghassi) [inline src/client/routes/BUILD.bazel]: Same as above, why does this need to be public?

File: src/server/BUILD.bazel [hunk 3]
```diff
@@ -1,12 +1,37 @@
...
+load("//bazel/js:defaults.bzl", "js_library", "nodejs_binary", "pkg_web")
+
+js_library(
+    name = "server",
+    srcs = [
+        ":getPageTemplate.js",
+        ":server.js",
+    ],
+    deps = [
+        "//src/client/routes:__pkg__",
```
Comment (Aghassi) [inline src/server/BUILD.bazel:11]: if this is why you need the other thing public, better to just state explicit dependency then expose it via public. Though it does sort of concern me that the server needs to rely on the routes to work, but maybe that's necessary.

File: src/server/server.js [hunk 4]
```diff
@@ -3,21 +3,61 @@
 const getPageTemplate = require("./getPageTemplate");
 
 const http = require("http");
+const path = require("path");
+const fs = require("fs");
+
+const config = require("@carto/routes");
```
Comment (Aghassi) [inline src/server/server.js]: Note where this comes from?

Comment (Aghassi) [reply src/server/server.js]: Also, does this only apply in the case where the server is unbundled? This makes the assumption we aren't running the server through a bundle I presume.

File: src/server/server.js [hunk 5]
```diff
@@ -3,21 +3,61 @@
...
-    })
-  );
+  console.log(`PATH REQUESTED: ${req.url}`);
+  if (req.url.startsWith("/favicon.ico")) {
+    res.writeHead(404);
+  } else if (req.url.startsWith("/__dev__/")) {
+    const filePath = req.url.replace("/__dev__/", "");
+    console.log(`ASSET REQUEST URL: ${filePath}`);
+
+    const fileToFetch = path.resolve("../..", "src/client/client", filePath);
```
Comment (Aghassi) [inline src/server/server.js]: this path seems brittle. Also the `client/client` can be stripped with `pkg_web` to make it `client/`

File: src/server/server.js [hunk 6]
```diff
@@ -3,21 +3,61 @@
...
+
+    console.log(fileToFetch);
+
+    res.writeHead(200);
+    res.end(fs.readFileSync(fileToFetch));
```
Comment (Aghassi) [inline src/server/server.js]: Is it ok that you are reading a file into a buffer here then returning the buffer?

Comment (sumwatshade) [reply src/server/server.js]: I can make sure it's cast to string here to be safe, but haven't seen issues here

File: src/server/server.js (hunk 4 above)
Comment (sumwatshade) [reply src/server/server.js]: The server always needs the route manifest, so this is true for both dev server and docker image. Will add comment to note the source

File: src/server/BUILD.bazel (hunk 3 above)
Comment (sumwatshade) [reply src/server/BUILD.bazel:11]: note, in this structure, only the dev server needs the routes to work, though these can be decoupled. Is it acceptable practice to run separate `ibazel` instances for two different parts of an app at the same time? like:

```bash
# terminal 1
./ibazel run //src/routes/secondary:dev

# terminal 2
./ibazel run //src/server:dev
```

Right now, only the latter is needed to run dev, and the docker image does not depend on routes

Comment (Aghassi) [reply src/server/BUILD.bazel:11]: That seems less than ideal? I think this is over optimizing at the moment, and for starters it's fine if we make the route config something that ends up in the docker container as part of the final product. But I expect that the dev server shouldn't even need the container. It's just a node server that looks somewhere in Bazel out for built artifacts and listens to the signals ibazel sends to know when to refresh the page.

**Pull request merged**